contrib/icu-cmake/CMakeLists.txt

```diff
@@ -346,115 +262,216 @@ set(ICUI18N_SOURCES
+"${ICU_SOURCE_DIR}/i18n/search.cpp"
+"${ICU_SOURCE_DIR}/i18n/selfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/sharedbreakiterator.cpp"
+"${ICU_SOURCE_DIR}/i18n/simpletz.cpp"
+"${ICU_SOURCE_DIR}/i18n/smpdtfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/smpdtfst.cpp"
+"${ICU_SOURCE_DIR}/i18n/sortkey.cpp"
+"${ICU_SOURCE_DIR}/i18n/standardplural.cpp"
+"${ICU_SOURCE_DIR}/i18n/string_segment.cpp"
+"${ICU_SOURCE_DIR}/i18n/strmatch.cpp"
+"${ICU_SOURCE_DIR}/i18n/strrepl.cpp"
+"${ICU_SOURCE_DIR}/i18n/stsearch.cpp"
+"${ICU_SOURCE_DIR}/i18n/taiwncal.cpp"
+"${ICU_SOURCE_DIR}/i18n/timezone.cpp"
+"${ICU_SOURCE_DIR}/i18n/titletrn.cpp"
+"${ICU_SOURCE_DIR}/i18n/tmunit.cpp"
+"${ICU_SOURCE_DIR}/i18n/tmutamt.cpp"
+"${ICU_SOURCE_DIR}/i18n/tmutfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/tolowtrn.cpp"
+"${ICU_SOURCE_DIR}/i18n/toupptrn.cpp"
+"${ICU_SOURCE_DIR}/i18n/translit.cpp"
+"${ICU_SOURCE_DIR}/i18n/transreg.cpp"
+"${ICU_SOURCE_DIR}/i18n/tridpars.cpp"
+"${ICU_SOURCE_DIR}/i18n/tzfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/tzgnames.cpp"
+"${ICU_SOURCE_DIR}/i18n/tznames.cpp"
+"${ICU_SOURCE_DIR}/i18n/tznames_impl.cpp"
+"${ICU_SOURCE_DIR}/i18n/tzrule.cpp"
+"${ICU_SOURCE_DIR}/i18n/tztrans.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucal.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucln_in.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucol.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucol_res.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucol_sit.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucoleitr.cpp"
+"${ICU_SOURCE_DIR}/i18n/ucsdet.cpp"
+"${ICU_SOURCE_DIR}/i18n/udat.cpp"
+"${ICU_SOURCE_DIR}/i18n/udateintervalformat.cpp"
+"${ICU_SOURCE_DIR}/i18n/udatpg.cpp"
+"${ICU_SOURCE_DIR}/i18n/ufieldpositer.cpp"
+"${ICU_SOURCE_DIR}/i18n/uitercollationiterator.cpp"
+"${ICU_SOURCE_DIR}/i18n/ulistformatter.cpp"
+"${ICU_SOURCE_DIR}/i18n/ulocdata.cpp"
+"${ICU_SOURCE_DIR}/i18n/umsg.cpp"
+"${ICU_SOURCE_DIR}/i18n/unesctrn.cpp"
+"${ICU_SOURCE_DIR}/i18n/uni2name.cpp"
 "${ICU_SOURCE_DIR}/i18n/units_complexconverter.cpp"
 "${ICU_SOURCE_DIR}/i18n/units_converter.cpp"
 "${ICU_SOURCE_DIR}/i18n/units_data.cpp"
-"${ICU_SOURCE_DIR}/i18n/units_router.cpp")
+"${ICU_SOURCE_DIR}/i18n/units_router.cpp"
+"${ICU_SOURCE_DIR}/i18n/unum.cpp"
+"${ICU_SOURCE_DIR}/i18n/unumsys.cpp"
+"${ICU_SOURCE_DIR}/i18n/upluralrules.cpp"
+"${ICU_SOURCE_DIR}/i18n/uregex.cpp"
+"${ICU_SOURCE_DIR}/i18n/uregexc.cpp"
+"${ICU_SOURCE_DIR}/i18n/uregion.cpp"
+"${ICU_SOURCE_DIR}/i18n/usearch.cpp"
+"${ICU_SOURCE_DIR}/i18n/uspoof.cpp"
+"${ICU_SOURCE_DIR}/i18n/uspoof_build.cpp"
+"${ICU_SOURCE_DIR}/i18n/uspoof_conf.cpp"
+"${ICU_SOURCE_DIR}/i18n/uspoof_impl.cpp"
+"${ICU_SOURCE_DIR}/i18n/utf16collationiterator.cpp"
+"${ICU_SOURCE_DIR}/i18n/utf8collationiterator.cpp"
+"${ICU_SOURCE_DIR}/i18n/utmscale.cpp"
+"${ICU_SOURCE_DIR}/i18n/utrans.cpp"
+"${ICU_SOURCE_DIR}/i18n/vtzone.cpp"
+"${ICU_SOURCE_DIR}/i18n/vzone.cpp"
+"${ICU_SOURCE_DIR}/i18n/windtfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/winnmfmt.cpp"
+"${ICU_SOURCE_DIR}/i18n/wintzimpl.cpp"
+"${ICU_SOURCE_DIR}/i18n/zonemeta.cpp"
+"${ICU_SOURCE_DIR}/i18n/zrule.cpp"
+"${ICU_SOURCE_DIR}/i18n/ztrans.cpp")
```

Comment:
> \# Note that we don't like any kind of binary plugins (because of runtime dependencies, vulnerabilities, ABI incompatibilities).
add_definitions(

Also `add_definitions` feels outdated to me.
But I'm not sure what features you need.
Here is my cmake

```
  add_definitions(-DU_DEBUG=1)
else ()
  add_definitions(-DU_DEBUG=0)
endif ()
add_definitions(
  -DU_NO_DEFAULT_INCLUDE_UTF_HEADERS=1
  -DU_DISABLE_RENAMING=0
  -DU_ENABLE_DYLOAD=0
  -DU_CHECK_DYLOAD=0

  -DUCONFIG_ENABLE_PLUGINS=0

  # Features
  -DUCONFIG_USE_ML_PHRASE_BREAKING=0
  -DUCONFIG_NO_LEGACY_CONVERSION=1
  -DUCONFIG_NO_CONVERSION=1
  -DUCONFIG_NO_FILE_IO=1
  -DUCONFIG_NO_MF2=1
  -DUCONFIG_NO_SERVICE=1
  -DUCONFIG_NO_IDNA=1
  -DUCONFIG_NO_FORMATTING=1
  -DUCONFIG_NO_TRANSLITERATION=0
  -DUCONFIG_NO_FILTERED_BREAK_ITERATION=1
  -DUCONFIG_NO_BREAK_ITERATION=0
  -DUCONFIG_NO_REGULAR_EXPRESSIONS=0
  -DUCONFIG_NO_NORMALIZATION=0
  -DUCONFIG_NO_COLLATION=0

  -DU_USING_ICU_NAMESPACE=1
  -DUNISTR_FROM_CHAR_EXPLICIT=explicit
  -DUNISTR_FROM_STRING_EXPLICIT=explicit
  -DU_NO_DEFAULT_INCLUDE_UTF_HEADERS=1
  -DU_HIDE_OBSOLETE_UTF_OLD_H=1
  -DU_STATIC_IMPLEMENTATION=1

  -D_REENTRANT 
)
```